### PR TITLE
zc - announcement page

### PR DESCRIPTION
**Pull request by imnotmomo**
## Overview
Announcement index page implemented for admin user, and has a Create Announcement button.

## Screenshots
Screenshot 2024-06-05 at 18 05 22

## Validation
https://happycows-qa.dokku-07.cs.ucsb.edu

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues
related to issue #4

### CODE REVIEW

**Review by jamesyfu — DISMISSED** [review]
LGTM!

**Review by YixinXue — DISMISSED** [review]
lgtm

**Review by pconrad — APPROVED** [review]
LGTM... will test on dokku

File: frontend/src/main/pages/AdminListAnnouncementsPage.js [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
\ No newline at end of file
+export default function AdminListAnnouncementsPage()
+{
+    const { commonsId } = useParams();
+    const { data: currentUser} = useCurrentUser();
+
+    // Stryker disable all
+    const { data: announcements, error: _error, status: _status } = useBackend(
+        [`/api/announcements/all?commonsId=${commonsId}`],
+        { method: "GET", url: `/api/announcements/all`, params: { commonsId } },
```
Comment (pconrad) [inline frontend/src/main/pages/AdminListAnnouncementsPage.js:16]: Check this against the URLs that are available on swagger.

I think you'll find that this URL doesn't exist.

**Review by pconrad — CHANGES_REQUESTED** [review]
I think there's a bug here.  Try fixing the url to make sure it matches the urls on swagger, deploy to dokku and test, and then let me know.

**Review by whuang602 — APPROVED** [review]
LGTM

**Comment by pconrad** [comment]
Points reverted because this feature was removed in PR #49

**Pull request merged**